docs/docs/adrs/adr-011-improving-test-confidence.md

```diff
@@ -0,0 +1,208 @@
+
+#### Main benefit
+MBT tooling can be used to generate test traces that can be executed by multiple different testing setups.
+
+### 2. Improve e2e tooling
+
+#### Matrix tests
+Instead of only running tests against current `main` branch we should adopt an approach where we also:
+- **run regression tests against released software versions** (`ICS v1/v2/v3`)
+- **run non-determinism tests to uncover issues quickly**
+
+Matrix tests can be implemented using [CometMock](https://github.com/informalsystems/CometMock) and refactoring our current e2e CI setup.
+
+##### Introducing e2e regression testing
+
+This e2e test suite would execute using a cronjob in our CI (nightly, multiple times a day etc.)
+
+
+Briefly, the same set of traces is run against different **maintained** versions of the software and the `main` branch.
+This would allow us to discover potential issues during development instead of in a testnet scenarios.
+
+The most valuable issues that can be discovered in this way are **state breaking changes**, **regressions** and **version incompatibilities**.
+
+
+The setup is illustrated by the image below.
+![e2e matrix tests](../../figures/matrix_e2e_tests.png)
+
+
+This table explains which versions are tested against each other for the same set of test traces:
+* ✅ marks a passing test
+* ❌ marks a failing test
+
+| **USES: ICS v1 PROVIDER**       | **start chain** | **add key** | **delegate** | **undelegate** | **redelegate** | **downtime** | **equivocation** | **stop chain** |
+|---------------------------------|-----------------|-------------|--------------|----------------|----------------|--------------|------------------|----------------|
+| **v1 consumer (sdk45,ibc4.3)**  | ✅               | ✅           | ✅            | ✅              | ✅              | ✅            | ✅                | ✅              |
+| **v2 consumer (sdk45, ibc4.4)** | ✅               | ✅           | ✅            | ✅              | ✅              | ✅            | ✅                | ✅              |
+| **v3 consumer (sdk47, ibc7)**   | ✅               | ✅           | ✅            | ✅              | ✅              | ✅            | ✅                | ✅              |
+| **main consumer**               | ❌               | ❌           | ❌            | ❌              | ❌              | ❌            | ❌                | ❌              |
+| **neutron**                     | ✅               | ✅           | ✅            | ✅              | ✅              | ✅            | ✅                | ❌              |
+| **stride**                      | ✅               | ✅           | ✅            | ✅              | ✅              | ✅            | ✅                | ❌              |
+
+
+##### Introducing e2e CometMock tests
+
+CometMock is a simplified version of the [CometBFT](https://github.com/cometbft/cometbft) consensus engine. It supports most operations performed by CometBFT while also being lightweight and relatively easy to use.
```

Comment:
nit: maybe replace "simplified" with "mocked implementation," since simplified might make someone think that CometMock is an actual consensus engine under the hoods.